app/jobs/save_original_to_s3_job.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class SaveOriginalToS3Job < ApplicationJob
+  queue_as :default
+
+  def default_priority
+    100
+  end
+
+  def perform(child_object_oid)
+    # check parent visibility is either public or YCO
+    child_object = ChildObject.find(child_object_oid)
+    parent_object = child_object.parent_object
+    if parent_object.visibility == "Private" || parent_object.visibility == "Redirect"
+      raise "Not copying image from #{parent_object.oid}. Parent object must have Public or Yale Community Only visibility."
```

Comment:
When it's not Public or YCO, it should just log and return from the job instead of raising the exception.   Similarly if the image is already on S3 (but don't log based on the ticket).  If it raises the exception, it will queue the job to retry.

Reply:
Just updated this with your suggested changes.  Thank you!